### PR TITLE
[rancher-alerting-drivers] configmap-reloader and PSP

### PR DESCRIPTION
**Pull request by jiaqiluo**
This PR contains the following changes:
- add supports for reloading the configMap in rancher-sachet
- add supports for pod security policy in rancher-sachet and rancher-prom2teams

related PR: https://github.com/rancher/charts/pull/1085
related issue: https://github.com/rancher/rancher/issues/29951

### CODE REVIEW

**Review by nickgerace — DISMISSED** [review]
LGTM. Using the `include` notation is interesting rather than `{{ .Release.Namespace }}` and related variables. Will have to look into it for other charts.

File: packages/rancher-sachet/charts/values.yaml [hunk 1]
```diff
@@ -10,6 +10,11 @@ global:
 nameOverride: "sachet"
 fullnameOverride: ""
 
+configReloader:
+  repository: rancher/mirrored-jimmidyson-configmap-reload
+  pullPolicy: IfNotPresent
+  tag: v0.4.0
```
Comment (cbron) [inline packages/rancher-sachet/charts/values.yaml:16]: confirmed this exists

File: packages/rancher-sachet/charts/templates/psp.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - min: 1
+        max: 65535
+  fsGroup:
+    rule: 'MustRunAs'
+    ranges:
+      - min: 1
+        max: 65535
+  readOnlyRootFilesystem: false
+  volumes:
+    - 'configMap'
```
Comment (cbron) [inline packages/rancher-sachet/charts/templates/psp.yaml:28]: Did this get tested in a hardened cluster or is it up to QA ?

File: packages/rancher-prom2teams/generated-changes/overlay/templates/role.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "prom2teams.fullname" . }}
```
Comment (cbron) [inline packages/rancher-prom2teams/generated-changes/overlay/templates/role.yaml:4]: I would make this specifically for the PSP, add `-psp`.

File: packages/rancher-sachet/charts/templates/psp.yaml (hunk 2 above)
Comment (jiaqiluo) [reply packages/rancher-sachet/charts/templates/psp.yaml:28]: I got the app installed on a hardened cluster successfully, but did not do any further test. We can leave it to QA.

File: packages/rancher-prom2teams/generated-changes/overlay/templates/role.yaml (hunk 3 above)
Comment (jiaqiluo) [reply packages/rancher-prom2teams/generated-changes/overlay/templates/role.yaml]: fixed.

**Review by cbron — DISMISSED** [review]
lgtm, but wait for @aiyengar2

**Review by aiyengar2 — DISMISSED** [review]
LGTM

**Review by aiyengar2 — APPROVED** [review]

**Pull request merged**